### ISSUE TITLE
Bleachbit-4.6.0 FreeBSD

### ISSUE THREAD

**Issue by snss806** [bleachbit/bleachbit#1589]
Hi!
I just downloaded bleachbit-4.6.0.zip, unpacked the tarball and run the following commands:
**make -C po local** 
**python3.9 bleachbit.py**

Running **python3.9 bleachbit.py**  is giving me the below errors:

> Traceback (most recent call last):                                                             │page](https://www.bleachbit.org)                                                              
>   File "/usr/home/user/Downloads/bleachbit-4.6.0/bleachbit.py", line 40, in <module>              │* [Support](https://www.bleachbit.org/help)                                                   
>     bleachbit.Unix.is_display_protocol_wayland_and_root_not_allowed()                          │* [Documentation](https://docs.bleachbit.org)                                                 
>   File "/usr/home/user/Downloads/bleachbit-4.6.0/bleachbit/Unix.py", line 746, in is_display_proto│* [Translate in Weblate](https://hosted.weblate.org/projects/bleachbit/): under evaluation    
> col_wayland_and_root_not_allowed                                                               │                                                                                              
>     bleachbit.Unix.is_linux_display_protocol_wayland() and                                     │                                                                                              
>   File "/usr/home/user/Downloads/bleachbit-4.6.0/bleachbit/Unix.py", line 730, in is_linux_display│## Licenses                                                                                   
> _protocol_wayland                                                                              │                                                                                              
>     assert(sys.platform.startswith('linux'))                                                   │BleachBit itself, including source code and cleaner definitions, is licensed under the [GNU Ge
> AssertionError "

**OS**: FreeBSD-13.2-RELEASE-p4

**Comment by az0** [comment]
@snss806 
Does [this patch](https://github.com/bleachbit/bleachbit/pull/1577) help?

**Comment by snss806** [comment]
@az0:

This is what I have now after re-downloading the zip file and running **python3.9. bleachbit.py**

```
Traceback (most recent call last):
  File "/usr/home/user/Downloads/test/bleachbit.py", line 40, in <module>
    bleachbit.Unix.is_display_protocol_wayland_and_root_not_allowed()
  File "/usr/home/user/Downloads/test/bleachbit/Unix.py", line 746, in is_display_protocol_wayland_and_root_not_allowed
    bleachbit.Unix.is_linux_display_protocol_wayland() and
  File "/usr/home/user/Downloads/test/bleachbit/Unix.py", line 730, in is_linux_display_protocol_wayland
    assert(sys.platform.startswith('linux'))
AssertionError
```

The result of **env |  grep XDG** command:
```
XDG_VTNR=9
XDG_DATA_DIRS=/usr/local/share:/usr/share
XDG_MENU_PREFIX=xfce-
XDG_SESSION_CLASS=user
XDG_SEAT=seat0
XDG_SESSION_TYPE=x11
XDG_RUNTIME_DIR=/var/run/user/1001
XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session3
XDG_CURRENT_DESKTOP=XFCE
XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0
XDG_CONFIG_DIRS=/etc/xdg:/usr/local/etc/xdg
```

**Comment by az0** [comment]
I commited a change to help: https://github.com/bleachbit/bleachbit/commit/0e710eb4186d967f048ed54bb6499b7469a26e0f . I'm considering a follow-up change to handle another scenario, so you can either test now or when that is ready.

**Comment by az0** [comment]
If you are able to test, try https://github.com/bleachbit/bleachbit/commit/15f8b0f7f86fdb61f4008f0d5e72968fb68046ac

**Comment by snss806** [comment]
@az0 :

After re-downloading the zip file and running **python3.9. bleachbit.py** the GUI launched without any issue prompting me to upgrade to Bleachbit 4.6.0. A quick inspection in Menu>About showed me that I'm running Bleachbit 4.6.1.
However, these messages appeared, both in terminal and in the GUI at the end of the clean process:

```
Special file type cannot be deleted: /home/user/.cache/at-spi/bus_0
Directory is not empty: /home/user/.cache/mozilla  <--#I assume that's because of having FF active
Directory is not empty: /home/user/.cache/at-spi
lsof: WARNING: compiled for FreeBSD release 13.1-RELEASE-p7; this is 13.2-RELEASE-p4.
Exception while running operation 'system': 'a bytes-like object is required, not 'str''
Traceback (most recent call last):
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/Worker.py", line 345, in run_operations
    for _dummy in self.clean_operation(operation):
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/Worker.py", line 159, in clean_operation
    for cmd in backends[operation].get_commands(option_id):
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/Cleaner.py", line 496, in get_commands
    is_open = FileUtilities.openfiles.is_open(path)
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/FileUtilities.py", line 174, in is_open
    self.scan()
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/FileUtilities.py", line 167, in scan
    for filename in open_files():
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/FileUtilities.py", line 134, in open_files
    for filename in files:
  File "/usr/home/user/Downloads/bleachbit-master460/bleachbit/FileUtilities.py", line 122, in open_files_lsof
    for f in run_lsof().split("\n"):
TypeError: a bytes-like object is required, not 'str'
```

Thank you for your hard work in dealing with these issues.

**Comment by az0** [comment]
That's helpful information. We fixed the original issue, so please open a new ticket for `TypeError: a bytes-like object is required`. This will make it easier to manage and triage

**Comment by snss806** [comment]
@az0 
Everything is working perfectly now. 
I don't think that I'll open, at least for now, another ticket for the `TypeError: a bytes-like object is required`. 
I think that this is because of FreeBSD and the repo you are using(not 100% sure): quarterly or latest.
What I've noticed was that changing from quarterly to latest or from latest to quarterly breaks Bleachbit and makes it unusable. 
I had to delete the .config/bleachbit folder and stick to latest. This fixed the issues.
I'll start testing bleachbit in FreeBSD14.0-RELEASE and let you know if there are any issues.
